packages/ui-components/src/components/Matrix/Pill.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { ReactNode } from 'react';
+import { Typography } from '@material-ui/core';
+import styled from 'styled-components';
+import { Tooltip } from '../Tooltip';
+
+const PillComponent = styled.div<{
+  $color?: string;
+}>`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 0.625rem;
+  min-width: 3.2rem;
+  padding-block: 0.35rem 0.25rem;
+  padding-inline: 0.7rem;
+  ${({ $color, theme }) =>
+    // apply a filter to the colour to make sure contrast with text is high enough. We do this with a linear gradient so that we can handle rgba, hex, and named colours
+    $color
+      ? `background: ${$color} linear-gradient(rgba(0,0,0,0.3),  rgba(0,0,0,0.3) 100%);`
```

Comment:
Good idea 🧠

I haven’t tried it locally, but I suspect using the overlay blend mode will produce slightly nicer results:

```css
background-blend-mode: normal, overlay;
```

Alternatively, depending on how much we trust that we will be provided relatively accessible colours, we could also wrap this style in `@media (prefer-contrast: more)`

Reply:
Just tried this - it looks exactly the same when applying `background-blend-mode`, so I am going to leave it out.
We also aren't confident the colours will be accessible, because it's a free for all basically. The main reason for adding a filter on top is a common use case is a traffic light system, which usually include yellow and of course, yellow background with white text is not ideal.